[func-count()] mo-server crashed in executing select count(*) from tbl_01 where a < 3

#### Can be reproduced ?
Yes

#### Steps:
create table tbl_01 (col1 int, a bigint unsigned, c char(10) not null);
insert into tbl_01 values (1,1,"a");
insert into tbl_01 values (2,2,"b");
insert into tbl_01 values (2,3,"c");
insert into tbl_01 values (3,4,"E");
insert into tbl_01 values (3,5,"C");
insert into tbl_01 values (3,6,"D");
select count(*) from tbl_01 where a < 3;

#### Expected behavior:
n/a

#### Actual behavior:
2021/09/14 18:23:58.467897 +0800 INFO cmd 3
2021/09/14 18:23:58.467996 +0800 INFO query:select count(*) from tbl_01 where a < 3
2021/09/14 18:23:58.469258 +0800 INFO time of Exec.Compile : 1.029ms
+++++++++
		:0 π(a,col1) -> σ(a < 3) -> γ([starCount(a) -> count(*)] -> =>
	:2 γ([sum(count(*)) -> count(*)] -> =>
:1 γ([sum(count(*)) -> count(*)] -> π(count(*)) -> sql output
+++++++++
2021/09/14 18:23:58.471129 +0800 INFO session connected {"session-id": 55, "addr": "127.0.0.1:61925"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x28 pc=0x103309518]

goroutine 45280 [running]:
matrixone/pkg/container/vector.(*Vector).Shuffle(0x0, {0x14002dc3e00, 0x2, 0x8}, 0x140005370e0)
	/Users/yangli/Documents/Origin/workspace/matrixone/pkg/container/vector/vector.go:507 +0x28
matrixone/pkg/sql/colexec/restrict.Call(0x140005370e0, {0x10453f5a0, 0x1400158e720})
	/Users/yangli/Documents/Origin/workspace/matrixone/pkg/sql/colexec/restrict/restrict.go:66 +0x8b0
matrixone/pkg/vm.Run({0x140004cb4a0, 0x4, 0x4}, 0x140005370e0)
	/Users/yangli/Documents/Origin/workspace/matrixone/pkg/vm/vm.go:49 +0xa4
matrixone/pkg/vm/pipeline.(*Pipeline).Run(0x14001551ee0, {0x1400076ab40, 0x1, 0x1}, 0x140005370e0)
	/Users/yangli/Documents/Origin/workspace/matrixone/pkg/vm/pipeline/pipeline.go:93 +0x6f8
matrixone/pkg/sql/compile.(*Scope).Run(0x14001d5d380, {0x104878b98, 0x14001dca1f8})
	/Users/yangli/Documents/Origin/workspace/matrixone/pkg/sql/compile/scope.go:77 +0x518
matrixone/pkg/sql/compile.(*Scope).MergeRun.func1({0x104878b98, 0x14001dca1f8}, 0x1400076ab10, 0x14001e98c00, 0x14001d5d380)
	/Users/yangli/Documents/Origin/workspace/matrixone/pkg/sql/compile/scope.go:92 +0x38
created by matrixone/pkg/sql/compile.(*Scope).MergeRun
	/Users/yangli/Documents/Origin/workspace/matrixone/pkg/sql/compile/scope.go:91 +0x200

#### Environment:
- Commit-id : fcc0b040075da8d86b7d2dbf9a91cabea5030ed8
- MySQL Client: mysql  Ver 8.0.26 for macos11.4 on arm64 (Homebrew)

#### Logs:

--- a/pkg/sql/build/group.go
+++ b/pkg/sql/build/group.go
@@ -33,6 +33,7 @@ func (b *build) buildGroupBy(o op.OP, ns tree.SelectExprs, grs tree.GroupBy, whe
 	var gs []*extend.Attribute
 	var es []aggregation.Extend
 
+	prev := o
 	{
 		var pes []*projection.Extend
 
@@ -93,7 +94,7 @@ func (b *build) buildGroupBy(o op.OP, ns tree.SelectExprs, grs tree.GroupBy, whe
 			es = append(es, aggregation.Extend{
 				Agg:   agg,
 				Alias: alias,
-				Name:  o.Columns()[0],
+				Name:  prev.Columns()[0],
 				Op:    aggregation.StarCount,
 			})
 		case *tree.UnresolvedName:

--- a/pkg/sql/build/summarize.go
+++ b/pkg/sql/build/summarize.go
@@ -55,6 +55,7 @@ func (b *build) buildSummarize(o op.OP, ns tree.SelectExprs, where *tree.Where) 
 	var fs []*tree.FuncExpr
 	var es []aggregation.Extend
 
+	prev := o
 	{
 		for _, n := range ns {
 			if !b.hasAggregate(n.Expr) {
@@ -106,7 +107,7 @@ func (b *build) buildSummarize(o op.OP, ns tree.SelectExprs, where *tree.Where) 
 			es = append(es, aggregation.Extend{
 				Agg:   agg,
 				Alias: alias,
-				Name:  o.Columns()[0],
+				Name:  prev.Columns()[0],
 				Op:    aggregation.StarCount,
 			})
 		case *tree.UnresolvedName:

--- a/pkg/sql/colexec/aggregation/max/float32.go
+++ b/pkg/sql/colexec/aggregation/max/float32.go
@@ -49,10 +49,10 @@ func (a *float32Max) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := max.Float32Max(vec.Col.([]float32))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v > a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/max/float64.go
+++ b/pkg/sql/colexec/aggregation/max/float64.go
@@ -49,10 +49,10 @@ func (a *float64Max) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := max.Float64Max(vec.Col.([]float64))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v > a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/max/int16.go
+++ b/pkg/sql/colexec/aggregation/max/int16.go
@@ -49,10 +49,10 @@ func (a *int16Max) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := max.Int16Max(vec.Col.([]int16))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v > a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/max/int32.go
+++ b/pkg/sql/colexec/aggregation/max/int32.go
@@ -49,10 +49,10 @@ func (a *int32Max) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := max.Int32Max(vec.Col.([]int32))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v > a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/max/int64.go
+++ b/pkg/sql/colexec/aggregation/max/int64.go
@@ -49,10 +49,10 @@ func (a *int64Max) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := max.Int64Max(vec.Col.([]int64))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v > a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/max/int8.go
+++ b/pkg/sql/colexec/aggregation/max/int8.go
@@ -49,10 +49,10 @@ func (a *int8Max) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := max.Int8Max(vec.Col.([]int8))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v > a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/max/uint16.go
+++ b/pkg/sql/colexec/aggregation/max/uint16.go
@@ -49,10 +49,10 @@ func (a *uint16Max) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := max.Uint16Max(vec.Col.([]uint16))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v > a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/max/uint32.go
+++ b/pkg/sql/colexec/aggregation/max/uint32.go
@@ -49,10 +49,10 @@ func (a *uint32Max) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := max.Uint32Max(vec.Col.([]uint32))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v > a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/max/uint64.go
+++ b/pkg/sql/colexec/aggregation/max/uint64.go
@@ -49,10 +49,10 @@ func (a *uint64Max) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := max.Uint64Max(vec.Col.([]uint64))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v > a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/max/uint8.go
+++ b/pkg/sql/colexec/aggregation/max/uint8.go
@@ -49,10 +49,10 @@ func (a *uint8Max) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := max.Uint8Max(vec.Col.([]uint8))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v > a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/min/float32.go
+++ b/pkg/sql/colexec/aggregation/min/float32.go
@@ -49,10 +49,10 @@ func (a *float32Min) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := min.Float32Min(vec.Col.([]float32))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v < a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/min/float64.go
+++ b/pkg/sql/colexec/aggregation/min/float64.go
@@ -49,10 +49,10 @@ func (a *float64Min) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := min.Float64Min(vec.Col.([]float64))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v < a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/min/int16.go
+++ b/pkg/sql/colexec/aggregation/min/int16.go
@@ -49,10 +49,10 @@ func (a *int16Min) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := min.Int16Min(vec.Col.([]int16))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v < a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/min/int32.go
+++ b/pkg/sql/colexec/aggregation/min/int32.go
@@ -49,10 +49,10 @@ func (a *int32Min) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := min.Int32Min(vec.Col.([]int32))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v < a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/min/int64.go
+++ b/pkg/sql/colexec/aggregation/min/int64.go
@@ -49,10 +49,10 @@ func (a *int64Min) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := min.Int64Min(vec.Col.([]int64))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v < a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/min/int8.go
+++ b/pkg/sql/colexec/aggregation/min/int8.go
@@ -49,10 +49,10 @@ func (a *int8Min) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := min.Int8Min(vec.Col.([]int8))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v < a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/min/uint16.go
+++ b/pkg/sql/colexec/aggregation/min/uint16.go
@@ -49,10 +49,10 @@ func (a *uint16Min) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := min.Uint16Min(vec.Col.([]uint16))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v < a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/min/uint32.go
+++ b/pkg/sql/colexec/aggregation/min/uint32.go
@@ -49,10 +49,10 @@ func (a *uint32Min) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := min.Uint32Min(vec.Col.([]uint32))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v < a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/min/uint64.go
+++ b/pkg/sql/colexec/aggregation/min/uint64.go
@@ -49,10 +49,10 @@ func (a *uint64Min) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := min.Uint64Min(vec.Col.([]uint64))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v < a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/min/uint8.go
+++ b/pkg/sql/colexec/aggregation/min/uint8.go
@@ -49,10 +49,10 @@ func (a *uint8Min) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		v := min.Uint8Min(vec.Col.([]uint8))
-		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 		if a.cnt == 0 || v < a.v {
 			a.v = v
 		}
+		a.cnt += int64(vec.Length() - vec.Nsp.Length())
 	}
 	return nil
 }

--- a/pkg/sql/colexec/aggregation/sum/uint.go
+++ b/pkg/sql/colexec/aggregation/sum/uint.go
@@ -55,13 +55,13 @@ func (a *uintSum) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		switch vec.Typ.Oid {
-		case types.T_int8:
+		case types.T_uint8:
 			a.sum += sum.Uint8Sum(vec.Col.([]uint8))
-		case types.T_int16:
+		case types.T_uint16:
 			a.sum += sum.Uint16Sum(vec.Col.([]uint16))
-		case types.T_int32:
+		case types.T_uint32:
 			a.sum += sum.Uint32Sum(vec.Col.([]uint32))
-		case types.T_int64:
+		case types.T_uint64:
 			a.sum += sum.Uint64Sum(vec.Col.([]uint64))
 		}
 		a.cnt += int64(vec.Length() - vec.Nsp.Length())

--- a/pkg/sql/colexec/aggregation/sum/uintSumCount.go
+++ b/pkg/sql/colexec/aggregation/sum/uintSumCount.go
@@ -54,13 +54,13 @@ func (a *uintSumCount) Fill(sels []int64, vec *vector.Vector) error {
 		a.cnt += int64(n - vec.Nsp.FilterCount(sels))
 	} else {
 		switch vec.Typ.Oid {
-		case types.T_int8:
+		case types.T_uint8:
 			a.sum += sum.Uint8Sum(vec.Col.([]uint8))
-		case types.T_int16:
+		case types.T_uint16:
 			a.sum += sum.Uint16Sum(vec.Col.([]uint16))
-		case types.T_int32:
+		case types.T_uint32:
 			a.sum += sum.Uint32Sum(vec.Col.([]uint32))
-		case types.T_int64:
+		case types.T_uint64:
 			a.sum += sum.Uint64Sum(vec.Col.([]uint64))
 		}
 		a.cnt += int64(vec.Length() - vec.Nsp.Length())

